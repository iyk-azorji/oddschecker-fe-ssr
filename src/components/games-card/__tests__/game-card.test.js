import React from "react";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router";
import GameCard from "../components/game-card";

describe("GameCard", () => {
  it("Matches the Snapshot", () => {
    const renderTree = renderer
      .create(
        <MemoryRouter initialEntries={["/"]}>
          <GameCard.WrappedComponent />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it("Renders blank when no props are passed", () => {
    const Wrapper = mount(
      <MemoryRouter initialEntries={["/"]}>
        <GameCard.WrappedComponent />
      </MemoryRouter>
    );
    Wrapper.mount();
  });
});
