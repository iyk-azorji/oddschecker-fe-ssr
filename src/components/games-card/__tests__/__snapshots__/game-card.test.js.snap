// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard Matches the Snapshot 1`] = `
<div
  className="sc-bdVaJa sc-bwzfXH game-card-container sc-htoDjs lopUrH"
>
  <a
    className="sc-chPdSV dFoQFp"
    href="undefinedgame/undefined"
    onClick={[Function]}
  >
    <div
      className="sc-bdVaJa sc-gqjmRU HaFtD"
    >
      <div
        className="sc-bdVaJa sc-bwzfXH sc-dnqmqq exaYQJ"
        width={
          Array [
            "5.5rem",
          ]
        }
      >
        <div
          className="sc-bdVaJa sc-iwsKbI izDLIk"
        >
          19:45
        </div>
        <div
          className="sc-bdVaJa sc-gZMcBi dknKRj"
        >
          26 Aug
        </div>
      </div>
      <div
        className="sc-jTzLTM ePrWel"
      >
        <div
          className="sc-cSHVUG fGvjcd"
        >
          <p
            className="sc-kAzzGY jLVifM"
          >
            Team 1
          </p>
        </div>
      </div>
      <div
        className="sc-bdVaJa sc-fjdhpX ctEBta"
      >
        <span>
          vs
        </span>
      </div>
      <div
        className="sc-jTzLTM ePrWel"
      >
        <div
          className="sc-cSHVUG fGvjcd"
        >
          <p
            className="sc-kAzzGY jLVifM"
          >
            Team 2
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
